### ISSUE TITLE
Stop using get_by!/2 in CodeCorps.ModelHelpers

### ISSUE THREAD

**Issue by joshsmith** [code-corps/code-corps-api#324]
When using `get_by!/2`, plug uses a different `conn` than our original one. This breaks CORS by not allowing us to render out our CORS headers.

While this behavior was nice to get started with, it is not tenable moving forward.

We can either solve this by using Canary's `load_resource` to render `json-api` errors where needed, or by rendering our own errors in the controller.

Either way, we need to switch to `get_by/2` and handle the if/else condition appropriately.

**Comment by JoshSmith** [comment]
`get!` located here:

```
test/controllers/preview_controller_test.exs:      preview = Preview |> Repo.get!(id)
test/controllers/preview_controller_test.exs:      preview = Preview |> Repo.get!(id)
test/controllers/project_skill_controller_test.exs:      project_skill = ProjectSkill |> Repo.get!(id)
test/controllers/role_skill_controller_test.exs:      role_skill = RoleSkill |> Repo.get!(id)
test/controllers/user_category_controller_test.exs:      user_category = UserCategory |> Repo.get!(id)
test/controllers/user_role_controller_test.exs:      user_role = UserRole |> Repo.get!(id)
test/controllers/user_skill_controller_test.exs:      user_skill = UserSkill |> Repo.get!(id)
web/controllers/comment_controller.ex:    comment = Repo.get!(Comment, id, preload: [:task])
web/controllers/comment_controller.ex:      |> Repo.get!(id)
web/controllers/organization_controller.ex:      |> Repo.get!(id)
web/controllers/organization_controller.ex:      |> Repo.get!(id)
web/controllers/project_controller.ex:      |> Repo.get!(id)
web/controllers/role_skill_controller.ex:      |> Repo.get!(id)
web/controllers/role_skill_controller.ex:    role_skill = Repo.get!(RoleSkill, id)
web/controllers/skill_controller.ex:      |> Repo.get!(id)
web/controllers/task_controller.ex:      |> Repo.get!(id)
web/controllers/task_controller.ex:      |> Repo.get!(id)
web/controllers/user_controller.ex:    user = User |> Repo.get!(id)
web/controllers/user_controller.ex:    changeset = User |> Repo.get!(id) |> User.update_changeset(Params.to_attributes(data))
```

**Comment by joshsmith** [comment]
`get_by!/2` here:

```
web/models/model_helpers.ex:    query |> CodeCorps.Repo.get_by!(slug: slug |> String.downcase)
```

**Comment by JoshSmith** [comment]
Will likely be closed by #332.
